### ISSUE TITLE
Have a question about the parameter "project" in YOLO26

### ISSUE THREAD

**Issue by husbarry** [ultralytics/ultralytics#23252]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Why can the parameter "project" in YOLO11 directly create a train folder under the initial directory, while in YOLO26, an intermediate folder "detect" is automatically created, along with a folder "predict".
<img width="198" height="204" alt="Image" src="https://github.com/user-attachments/assets/affe8bcf-d1ab-4380-9c17-594fa9bcbead" />
<img width="225" height="254" alt="Image" src="https://github.com/user-attachments/assets/6ed08659-fe72-48ea-9331-cdb7c6c9159b" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @husbarry, thank you for your interest in Ultralytics 🚀! This is an automated response to help move things along quickly, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (including questions about run output directories like `project/`, `name`, and auto-created subfolders), please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). 🙏

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You should post the Ultralytics versions. There's no YOLO11 or YOLO26 repo. You're using Ultralytics which is a library that supports all the models. It's not dedicated to a particular model.

**Comment by Y-T-G** [comment]
Also what's the command you're using?

**Comment by cjxxu** [comment]
I met the same problem. there are many empty "predict" files created when predict.

<img width="285" height="745" alt="Image" src="https://github.com/user-attachments/assets/d8de27e3-4fd0-4154-98d9-33f61fd97b19" />

**Comment by xavan83** [comment]
8.3.x version creates yolo11/train+index
8.4.x version creates only runsyolo (without anything else)

train parameters in python: save_dir='runsyolo',  project='yolo11', 

something changes between the 2versions

**Comment by Y-T-G** [comment]
@xavan83 There are 253 minor versions with 8.3. Which one are you referring to? The behavior between 8.3.253 and 8.4.0 is the same.

**Comment by Y-T-G** [comment]
@xavan83 `save_dir` overrides `project` and `name`. You should not pass `save_dir` if you want it to use `project` and `name` to create folder

**Comment by xavan83** [comment]
Using only project "restores" the old behavior (1directory + train, train1, train2)
I was using mainly all 8.3.x versions since last 6months. (i.e. 8.3.243, 8.3.244,etc)

**Comment by husbarry** [comment]
> You should post the Ultralytics versions. There's no YOLO11 or YOLO26 repo. You're using Ultralytics which is a library that supports all the models. It's not dedicated to a particular model.
#
The code version I am using is v8.4.0. When executing the training task, I specified the parameters project='runs/train' and name='exp', but the resulting args.yaml file shows the parameter save_dir as 'E:\Code\PyCharm\YOLO26\ultralytics-main\runs\detect\runs\train\exp'. An additional 'runs/detect' folder was automatically added, and an empty folder named 'predict' was also created under the 'detect' folder. I am also unclear about the purpose of this folder, just like this user.

> I met the same problem. there are many empty "predict" files created when predict.
> 
> <img alt="Image" width="285" height="745" src="https://private-user-images.githubusercontent.com/38255733/536497075-d8de27e3-4fd0-4154-98d9-33f61fd97b19.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg1NDUwMzksIm5iZiI6MTc2ODU0NDczOSwicGF0aCI6Ii8zODI1NTczMy81MzY0OTcwNzUtZDhkZTI3ZTMtNGZkMC00MTU0LTk4ZDktMzNmNjFmZDk3YjE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTE2VDA2MjUzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0ZWIzZDQ0M2VmN2MzNmZmZDllZWQ4MjczMjcwZTljMTI0YmRmNWJiZjJkYjk4OGJiZWUzZTgzNDM1Y2ViYjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0U8ArTqi-S5KKkCx4zdrdO5sj3HWGiQi8jptXQinK_s">

**Comment by litengfeigithub** [comment]
<img width="1430" height="178" alt="Image" src="https://github.com/user-attachments/assets/51182cdd-1027-459c-9e69-794237e283f5" />
I encountered this issue as well. When I was performing segmentation training, a detect folder and a predict folder under it were automatically generated.
The save path I set is "runs/seg..." but a "segment" folder is automatically generated at the front of the directory

**Comment by glenn-jocher** [comment]
This is expected from how `save_dir` is computed: Ultralytics builds outputs under `runs_dir/<task>/...`, and if `project` contains multiple path parts (e.g. `runs/train`), it’s treated as “nested” and gets appended under that task root, which is why you see `runs/detect/runs/train/exp` (same for `segment`). See the implementation in [`ultralytics.cfg.get_save_dir`](https://docs.ultralytics.com/reference/cfg/__init__/#ultralytics.cfg.get_save_dir).

If you want `runs/train/exp` without the extra `detect` prefix, pass a single-part `project` and put the subfolders in `name`, e.g.:

```python
model.train(data="...", project="runs", name="train/exp")
```

The empty `predict*` folders are created when `predict()` runs (including implicit prediction via `model(source)`), since the default run name is the mode (`predict`) and it increments each time; set `name=...` with `exist_ok=True` and/or `save=False` to avoid creating new run dirs. If you want to change the global base directory instead, use the documented [Ultralytics settings for `runs_dir`](https://docs.ultralytics.com/quickstart/#ultralytics-settings).
